File: client_app/android/app/src/main/java/com/aionsigma_app/room/UserInfoDao.java
package com.aionsigma_app.room;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface UserInfoDao {
    @Query("SELECT * FROM userinfos")
    List<UserInfo> getAll();

    @Query("SELECT * FROM userinfos WHERE userInfoId IN (:userInfoIds)")
    List<UserInfo> loadAllByIds(String[] userInfoIds);

    @Query("UPDATE userinfos SET synced = :now WHERE userInfoId IN (:userInfoIds)")
    void updateSynced(String[] userInfoIds, long now);

    @Query("SELECT * FROM userinfos WHERE synced > 0 Order by createdAt asc LIMIT :limit")
    List<UserInfo> getAllForSync(int limit);

    @Insert
    void insert(UserInfo... userInfo);

    @Query("Delete From userInfos Where synced is not null")
    void delete();

}

File: android/AionSigma/app/src/main/java/com/aionsigma/android/Room/UserInfoDao.kt
package com.aionsigma.android.Room

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import io.reactivex.Flowable

@Dao
interface UserInfoDao {
    @Query("SELECT * FROM userinfos")
    fun getAll(): Flowable<List<UserInfo>>

    @Query("SELECT * FROM userinfos WHERE userInfoId IN (:userInfoIds)")
    fun loadAllByIds(userInfoIds: Array<String>): List<UserInfo>

    @Query("UPDATE userinfos SET synced = :now WHERE userInfoId IN (:userInfoIds)")
    fun updateSynced(userInfoIds: Array<String>, now: Long)

    @Query("SELECT * FROM userinfos WHERE synced > 0 Order by createdAt asc LIMIT :limit")
    fun getAllForSync(limit: Int): List<UserInfo>

    @Insert
    fun insert(userInfo: UserInfo)

    @Query("Delete From userInfos Where synced is not null")
    fun delete()

}